server/v2/stf/stf.go

```diff
@@ -42,29 +44,39 @@ type STF[T transaction.Tx] struct {
 
 // NewSTF returns a new STF instance.
 func NewSTF[T transaction.Tx](
-	handleMsg func(ctx context.Context, msg transaction.Msg) (transaction.Msg, error),
-	handleQuery func(ctx context.Context, req transaction.Msg) (transaction.Msg, error),
+	msgRouterBuilder *MsgRouterBuilder,
```

Comment:
nit, can we add the logger here so it closes this as well: https://github.com/cosmos/cosmos-sdk/issues/20663